android/KMEA/app/src/main/java/com/keyman/engine/KeyboardPickerActivity.java

```diff
@@ -252,7 +253,11 @@ protected void onResume() {
       adapter.notifyDataSetChanged();
     }
 
-    int curKbPos = KeyboardController.getInstance().getKeyboardIndex(KMKeyboard.currentKeyboard());
+
+    SharedPreferences prefs = this.getSharedPreferences(this.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
```

Comment:
prefs are only used if the currentKeyboard is null, so I would not initialize this unless it is null
Instead of the ternary operator, just use a regular if-else, so the logic is easier to follow, and we only load variables that we are going to use.